src/init/features/functions/index.ts

```diff
@@ -30,6 +26,106 @@ export async function doSetup(setup: any, config: any, options: Options) {
       ensure(projectId, "runtimeconfig.googleapis.com", "unused", true),
     ]);
   }
+  setup.functions = {};
+  if (!config.src.functions) {
+    // if functions have not been initialized yet
+    setup.config.functions = [];
+    return codebaseSetup(setup, config, false);
+  }
+  // if functions have already been initialized, make sure config is validated and normalized
+  setup.config.functions = normalizeAndValidate(setup.config.functions);
+
+  logger.info();
+  logger.info("Detected " + clc.bold("existing codebase(s)."));
+  logger.info();
+
+  const choices = [
+    {
+      name: "Initialize",
+      value: "new",
+    },
+    {
+      name: "Re-initialize",
+      value: "reinit",
+    },
+  ];
+  const initOpt = await promptOnce({
+    type: "list",
+    message: "Would you like to initialize a new codebase, or reinitialize an existing one?",
+    default: "new",
+    choices,
+  });
+  return codebaseSetup(setup, config, initOpt === "reinit");
+}
+
+/**
+ *  User dialogue to set up configuration for functions codebase.
+ */
+async function codebaseSetup(setup: any, config: Config, reinit: boolean): Promise<any> {
```

Comment:
nit* Just to offer different perspective - I don't see why we wouldn't define 2 different functions (`reinitCodebase` and `setupNewCodebase`) now that there is very little shared code between when `reinit=true` and `reinit=false`.

Reply:
fair point!